red-gate/SQLCover
# Fix/TBE-46 - Change packages for SQL Test support for SSMS 19 GA

Change Microsoft.SqlServer.DacFx.x64 package to Microsoft.SqlServer.TransactSql.ScriptDom.ReSigned and change .NET TargetFrameworkVersion from 4.5.2 to 4.7.2 and add Parser for SSMS19.

Fixes # .

Changing the Microsoft.SqlServer.DacFx package because it is closely related to the Microsoft.SqlServer.TransactSql.ScriptDom version, which does not allow you to work with the available version of this package in different versions of SSMS.

- DacFx.x64 v140 uses ScriptDom v14 available in SSMS17
- DacFx.x64 v150 uses ScriptDom v15 available in SSMS18
- DacFx v160 uses ScriptDom v16 available in SSMS19

In the case of SQL Test, this causes that the Code coverage tab, when trying to load a log from SSMS, will work only for the version for which the DacFx version uses the same version of the ScriptDom package as SSMS.

Using Microsoft.SqlServer.TransactSql.ScriptDom.Resigned allows you to workaround this problem by using SQLCover's own ScriptDom package

Changes proposed in this pull request:
 -  .NET TargetFrameworkVersion from 4.5.2 to 4.7.2
 -  Microsoft.SqlServer.DacFx.x64 package to Microsoft.SqlServer.TransactSql.ScriptDom.ReSigned
 -  Add Parser for SSMS19 and set default Parser

How to test this code:
 - Manual tests for SQL Test 4 in SSMS17, SSMS18 and SSMS19
 - Download the SQLTest repository from the fix/TBE-46 branch and SQLCover from the fix/TBE-46 branch --> build the SQLCover project --> from the SQL Test project remove the reference to the SQLCover package --> Include the SQLCover project from the SQLCover solution to the SQL Test solution --> add references to the added SQLCover project in the SQL Test project and build solutions --> Install SSMS version 17/18/19 GA and install the SQL Test 4 add-on --> replace the dll file path for SQL Test from the original to the one created after build the SQL Test solution --> run SSMS --> connect to any server --> run SQL Test --> connect any database --> install the tSQLt add-on --> run tests for the selected database --> in the right In the window, go to the Code coverage tab and wait for the result - the response should be the system log from SSMS

Has been tested on (remove any that don't apply):
 - SQL Server 2017
 - SQL Server 2018
 - SQL Server 2019
